src/V3Tristate.cpp

```diff
@@ -636,6 +665,117 @@ class TristateVisitor final : public TristateBaseVisitor {
+        return assignmentOfValueOnAllBits(*maxIt, value) ? *maxIt : nullptr;
+    }
+
+    void removeWeakerAssignments(Assigns& assigns) {
+        // Weaker assignments are these assignments that can't change the final value of the net.
+        // If we know the value of the RHS, we take into account only strength corresponding to its
+        // value. We skip here assignments of constants that have bits both 0 and 1, because it
+        // would involve handling parts of bits separately.
+
+        // First we want to find the strongest assignment that has value on the RHS consisting of
+        // only 1 or only 0.
+        AstAssignW* const strongest0p = getStrongestAssignmentOfValue(assigns, 0);
+        AstAssignW* const strongest1p = getStrongestAssignmentOfValue(assigns, 1);
+        AstAssignW* strongestp = nullptr;
+        uint8_t greatestKnownStrength = 0;
+        const auto getIfStrongest = [&](AstAssignW* const strongestCandidatep, bool value) {
+            if (!strongestCandidatep) return;
+            uint8_t strength = getStrength(strongestCandidatep, value);
+            if (strength >= greatestKnownStrength) {
+                greatestKnownStrength = strength;
+                strongestp = strongestCandidatep;
+            }
+        };
+        getIfStrongest(strongest0p, 0);
+        getIfStrongest(strongest1p, 1);
+
+        if (strongestp) {
+            // Then we can safely remove all weaker than it
+            // We also remove assignments of the same strength, because we don't need duplicates
+            // One problem is with 2 assignments of different values and equal strengths. It should
+            // result in assignment of x value, but these values aren't supported now.
+            auto removedIt
+                = std::remove_if(assigns.begin(), assigns.end(), [&](AstAssignW* assignp) {
+                      if (assignp == strongestp) return false;
+                      uint8_t strength0 = getStrength(assignp, 0);
+                      uint8_t strength1 = getStrength(assignp, 1);
+                      bool toRemove = (strength0 <= greatestKnownStrength
+                                       && strength1 <= greatestKnownStrength)
+                                      || (strength0 <= greatestKnownStrength
+                                          && assignmentOfValueOnAllBits(assignp, 0))
+                                      || (strength1 <= greatestKnownStrength
+                                          && assignmentOfValueOnAllBits(assignp, 1));
+
+                      if (toRemove) {
+                          // We don't want to propagate tristate if its assignment is removed.
+                          TristateVertex* const vertexp
+                              = reinterpret_cast<TristateVertex*>(assignp->rhsp()->user5p());
+                          if (vertexp) vertexp->isTristate(false);
+                          assignp->unlinkFrBack()->deleteTree();
```

Comment:
```suggestion
                            VL_DO_DANGLING(pushDeletep(assignp->unlinkFrBack(), assignp);
```

Easier to debug if delete later.